src/com/google/enterprise/adaptor/database/ResponseGenerator.java

```diff
@@ -386,20 +404,126 @@ public void generateResponse(ResultSet rs, Response resp)
         throws IOException, SQLException {
       overrideContentType(rs, resp);
       overrideDisplayUrl(rs, resp);
-      Blob blob = rs.getBlob(getContentColumnName());
-      InputStream in = blob.getBinaryStream();
+
+      ResultSetMetaData rsMetaData = rs.getMetaData();
+      int index = rs.findColumn(getContentColumnName());
+      int columnType = rsMetaData.getColumnType(index);
+      log.log(Level.FINEST, "Content column name: {0}, Type: {1}",
+          new Object[] {getContentColumnName(), columnType});
+
       OutputStream out = resp.getOutputStream();
-      com.google.enterprise.adaptor.IOHelper.copyStream(in, out);
-      in.close();
-      if (!(blob instanceof javax.sql.rowset.serial.SerialBlob)) {
-        // SerialBlob is adamant about not supporting free
-        try {
-          blob.free();
-        } catch (java.sql.SQLFeatureNotSupportedException | 
-            UnsupportedOperationException unsupported) {
-          // let JVM garbage collection deal with it
-        }
+      switch (columnType) {
+        case Types.LONGVARCHAR:
```

Comment:
The order here is kind of random. Let's put like together with like, and in something like the order most likely to be used: BLOB, CLOB, NCLOB, LONGVARBINARY, LONGVARCHAR, LONGNVARCHAR, BINARY + VARBINARY, string, default.